xsqlite: symbol lookup error

### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The most recent xsqlite version that is currently installable from conda-forge seems to be broken,

```
(base) root@8cca2c0c3f67:/# conda create -n minimal -c conda-forge xeus-sqlite
(base) root@8cca2c0c3f67:/# conda activate minimal                                                                                                                                                            
(minimal) root@8cca2c0c3f67:/#  xsqlite                                                                                                                                                                       
xsqlite: symbol lookup error: /miniconda3/envs/minimal/bin/../lib/libxeus-sqlite.so.0: undefined symbol: _ZN4xeus12xinterpreter21internal_request_implERKN8nlohmann10basic_jsonISt3mapSt6vectorNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEblmdSaNS1_14adl_serializerES4_IhSaIhEEEE  
```

which also shows up in Jupyter server logs,

```
 # [I 2023-03-16 14:00:08.772 ServerApp] AsyncIOLoopKernelRestarter: restarting kernel (1/5), new random ports
 # /miniconda3/bin/xsqlite: symbol lookup error: /miniconda3/bin/../lib/libxeus-sqlite.so.0: undefined symbol: _ZN4xeus12xinterpreter21internal_request_implERKN8nlohmann10basic_jsonISt3mapSt6vectorNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEblmdSaNS1_14adl_serializerES4_IhSaIhEEEE
 # only install libspatialite and install xeus-sqlite from scratch
```

when attempting to start the corresponding Jupyter kernel (not installed in the above conda environment to keep things minimal).

Since I needed the package today for compatibility tests with a particular SQLite extension library, I ended up compiling `xeus-sqlite` myself (using [this particular list](https://github.com/conda-forge/xeus-sqlite-feedstock/blob/5a0bf1483c5fe822b2f6137a74b2bfc9de0ca1f4/recipe/meta.yaml#L17-L32) as build environment) which unfortunately failed with,

```
(minimal) root@8cca2c0c3f67:/# mamba create -n compile1 -c conda-forge cxx-compiler cmake make 'xeus-zmq >=1.0.2,<2.0.0' cppzmq zeromq nlohmann_json xtl 'cpp-tabulate >=1.3,<2' sqlitecpp sqlite 'xvega >=0.0.10' 'xvega-bindings >=0.0.10' 'xproperty >=0.11.0,<0.12.0'
(minimal) root@8cca2c0c3f67:/# conda activate compile1
(compile1) root@8cca2c0c3f67:/
(compile1) root@8cca2c0c3f67:/# wget https://github.com/jupyter-xeus/xeus-sqlite/archive/refs/tags/0.6.0.zip
(compile1) root@8cca2c0c3f67:/# unzip 0.6.0.zip
(compile1) root@8cca2c0c3f67:/# cd xeus-sqlite-0.6.0/
(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0# cmake -D CMAKE_INSTALL_PREFIX=$CONDA_PREFIX
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.
-- The C compiler identification is GNU 11.3.0
-- The CXX compiler identification is GNU 11.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /miniconda3/envs/compile1/bin/x86_64-conda-linux-gnu-cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /miniconda3/envs/compile1/bin/x86_64-conda-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building xeus-sqlite v0.6.0
-- Found nlohmann_json: /miniconda3/envs/compile1/share/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found suitable version "3.11.2", minimum required is "3.2.0") 
-- Found LibUUID: /miniconda3/envs/compile1/lib/libuuid.so  
-- Found SQLite3: /miniconda3/envs/compile1/include (found version "3.40.0") 
-- Found xvega-bindings: /miniconda3/envs/compile1/lib/cmake/xvega-bindings/xvega-bindingsConfig.cmake (found suitable version "0.0.10", minimum required is "0.0.3") 
-- Found nlohmann_json: /miniconda3/envs/compile1/share/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found suitable version "3.11.2", minimum required is "3.7.3") 
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Performing Test HAS_CPP17_FLAG
-- Performing Test HAS_CPP17_FLAG - Success
-- Found OpenSSL: /miniconda3/envs/compile1/lib/libcrypto.so (found version "3.1.0")  
-- Configuring done (0.7s)
-- Generating done (0.0s)
-- Build files have been written to: /xeus-sqlite-0.6.0
(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0# make
[ 10%] Building CXX object CMakeFiles/xeus-sqlite.dir/src/xeus_sqlite_interpreter.cpp.o
/xeus-sqlite-0.6.0/src/xeus_sqlite_interpreter.cpp: In member function 'void xeus_sqlite::interpreter::process_SQLite_input(int, std::unique_ptr<SQLite::Database>&, const string&, xv::df_type&)':
/xeus-sqlite-0.6.0/src/xeus_sqlite_interpreter.cpp:323:33: error: cannot convert 'std::vector<std::variant<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, tabulate::Table> >' to 'const Row_t&' {aka 'const std::vector<std::variant<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, std::basic_string_view<char, std::char_traits<char> >, tabulate::Table> >&'}
  323 |             plain_table.add_row(col_names);
      |                                 ^~~~~~~~~
      |                                 |
      |                                 std::vector<std::variant<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, tabulate::Table> >
In file included from /xeus-sqlite-0.6.0/src/xeus_sqlite_interpreter.cpp:21:
/miniconda3/envs/compile1/include/tabulate/table.hpp:65:31: note:   initializing argument 1 of 'tabulate::Table& tabulate::Table::add_row(const Row_t&)'
   65 |   Table &add_row(const Row_t &cells) {
      |                  ~~~~~~~~~~~~~^~~~~
/xeus-sqlite-0.6.0/src/xeus_sqlite_interpreter.cpp:359:37: error: cannot convert 'std::vector<std::variant<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, tabulate::Table> >' to 'const Row_t&' {aka 'const std::vector<std::variant<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, std::basic_string_view<char, std::char_traits<char> >, tabulate::Table> >&'}
  359 |                 plain_table.add_row(row);
      |                                     ^~~
      |                                     |
      |                                     std::vector<std::variant<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char*, tabulate::Table> >
In file included from /xeus-sqlite-0.6.0/src/xeus_sqlite_interpreter.cpp:21:
/miniconda3/envs/compile1/include/tabulate/table.hpp:65:31: note:   initializing argument 1 of 'tabulate::Table& tabulate::Table::add_row(const Row_t&)'
   65 |   Table &add_row(const Row_t &cells) {
      |                  ~~~~~~~~~~~~~^~~~~
make[2]: *** [CMakeFiles/xeus-sqlite.dir/build.make:76: CMakeFiles/xeus-sqlite.dir/src/xeus_sqlite_interpreter.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/xeus-sqlite.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0#
```

Digging into details I found out that the problem seems to be caused by the `cpp-tabulate` package version (see potentially related problem here: https://github.com/p-ranav/tabulate/issues/101). I was able to successfully compile `xeus-sqlite` by downgrading `cpp-tabulate`,

```
(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0# mamba install -c conda-forge 'cpp-tabulate >=1.3,<1.5'
(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0# cmake -D CMAKE_INSTALL_PREFIX=$CONDA_PREFIX
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.
-- Building xeus-sqlite v0.6.0
-- Found nlohmann_json: /miniconda3/envs/compile1/share/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found suitable version "3.11.2", minimum required is "3.2.0") 
-- Found nlohmann_json: /miniconda3/envs/compile1/share/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found suitable version "3.11.2", minimum required is "3.7.3") 
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /xeus-sqlite-0.6.0
(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0# make
[ 10%] Building CXX object CMakeFiles/xeus-sqlite.dir/src/xeus_sqlite_interpreter.cpp.o
[ 20%] Building CXX object CMakeFiles/xeus-sqlite.dir/src/xvega_sqlite.cpp.o
[ 30%] Building CXX object CMakeFiles/xeus-sqlite.dir/src/xlite.cpp.o
[ 40%] Linking CXX shared library libxeus-sqlite.so
[ 40%] Built target xeus-sqlite
[ 50%] Building CXX object CMakeFiles/xeus-sqlite-static.dir/src/xeus_sqlite_interpreter.cpp.o
[ 60%] Building CXX object CMakeFiles/xeus-sqlite-static.dir/src/xvega_sqlite.cpp.o
[ 70%] Building CXX object CMakeFiles/xeus-sqlite-static.dir/src/xlite.cpp.o
[ 80%] Linking CXX static library libxeus-sqlite.a
[ 80%] Built target xeus-sqlite-static
[ 90%] Building CXX object CMakeFiles/xsqlite.dir/src/main.cpp.o
[100%] Linking CXX executable xsqlite
[100%] Built target xsqlite
(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0# make install
[ 40%] Built target xeus-sqlite
[ 80%] Built target xeus-sqlite-static
[100%] Built target xsqlite
Install the project...
-- Install configuration: "RELEASE"
-- Installing: /miniconda3/envs/compile1/lib/libxeus-sqlite.so.0.6.0
-- Installing: /miniconda3/envs/compile1/lib/libxeus-sqlite.so.0
-- Installing: /miniconda3/envs/compile1/lib/libxeus-sqlite.so
-- Installing: /miniconda3/envs/compile1/include/xeus-sqlite/xeus_sqlite_config.hpp
-- Installing: /miniconda3/envs/compile1/include/xeus-sqlite/xeus_sqlite_interpreter.hpp
-- Installing: /miniconda3/envs/compile1/include/xeus-sqlite/xvega_sqlite.hpp
-- Installing: /miniconda3/envs/compile1/lib/libxeus-sqlite.a
-- Up-to-date: /miniconda3/envs/compile1/include/xeus-sqlite/xeus_sqlite_config.hpp
-- Up-to-date: /miniconda3/envs/compile1/include/xeus-sqlite/xeus_sqlite_interpreter.hpp
-- Up-to-date: /miniconda3/envs/compile1/include/xeus-sqlite/xvega_sqlite.hpp
-- Installing: /miniconda3/envs/compile1/bin/xsqlite
-- Installing: /miniconda3/envs/compile1/share/jupyter/kernels
-- Installing: /miniconda3/envs/compile1/share/jupyter/kernels/xsqlite
-- Installing: /miniconda3/envs/compile1/share/jupyter/kernels/xsqlite/logo-32x32.png
-- Installing: /miniconda3/envs/compile1/share/jupyter/kernels/xsqlite/logo-64x64.png
-- Installing: /miniconda3/envs/compile1/share/jupyter/kernels/xsqlite/kernel.json
-- Installing: /miniconda3/envs/compile1/lib/cmake/xeus-sqlite/xeus-sqliteConfig.cmake
-- Installing: /miniconda3/envs/compile1/lib/cmake/xeus-sqlite/xeus-sqliteConfigVersion.cmake
-- Installing: /miniconda3/envs/compile1/lib/cmake/xeus-sqlite/xeus-sqliteTargets.cmake
-- Installing: /miniconda3/envs/compile1/lib/cmake/xeus-sqlite/xeus-sqliteTargets-release.cmake
(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0# 
```

The xsqlite command then works as expected,

```
(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0# which xsqlite
/miniconda3/envs/compile1/bin/xsqlite
(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0# xsqlite
registering handler for SIGSEGV
Starting xeus-sqlite kernel...

If you want to connect to this kernel from an other client, just copy and paste the following content inside of a `kernel.json` file. And then run for example:

# jupyter console --existing kernel.json

kernel.json

{
    "transport": "tcp",
    "ip": "127.0.0.1",
    "control_port": 59160,
    "shell_port": 58035,
    "stdin_port": 57773,
    "iopub_port": 64823,
    "hb_port": 49944,
    "signature_scheme": "hmac-sha256",
    "key": "93cee6f5c9e34383947a8b7e720880ed"
}

Run with XEUS 3.0.5

^C(compile1) root@8cca2c0c3f67:/xeus-sqlite-0.6.0# 
```

both from the command line and from JupyterLab directly.

### Installed packages

```shell
# packages in environment at /miniconda3/envs/minimal:                                                                                                                                                        
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
ca-certificates           2022.12.7            ha878542_0    conda-forge
libgcc-ng                 12.2.0              h65d4601_19    conda-forge
libgomp                   12.2.0              h65d4601_19    conda-forge
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libsqlite                 3.40.0               h753d276_0    conda-forge
libstdcxx-ng              12.2.0              h46fd767_19    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libzlib                   1.2.13               h166bdaf_4    conda-forge
ncurses                   6.3                  h27087fc_1    conda-forge
openssl                   3.1.0                h0b41bf4_0    conda-forge
readline                  8.1.2                h0f457ee_0    conda-forge
sqlite                    3.40.0               h4ff8645_0    conda-forge
sqlitecpp                 3.1.1                h0251d09_0    conda-forge
xeus                      3.0.5                hac2b420_1    conda-forge
xeus-sqlite               0.6.0                hdd8d261_0    conda-forge
xeus-zmq                  1.0.2                h0541b36_2    conda-forge
xvega                     0.0.10               hf52228f_1    conda-forge
xvega-bindings            0.0.10               hf52228f_1    conda-forge
zeromq                    4.3.4                h9c3ff4c_1    conda-forge
```


### Environment info

```shell
active environment : minimal
    active env location : /miniconda3/envs/minimal
            shell level : 2
       user config file : /root/.condarc
 populated config files : 
          conda version : 23.1.0
    conda-build version : not installed
         python version : 3.10.9.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.35=0
                          __linux=5.4.0=0
                          __unix=0=0
       base environment : /miniconda3  (writable)
      conda av data dir : /miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /miniconda3/pkgs
                          /root/.conda/pkgs
       envs directories : /miniconda3/envs
                          /root/.conda/envs
               platform : linux-64
             user-agent : conda/23.1.0 requests/2.28.1 CPython/3.10.9 Linux/5.4.0-144-generic ubuntu/22.04.2 glibc/2.35
                UID:GID : 0:0
             netrc file : None
           offline mode : False
```

--- a/.scripts/build_steps.sh
+++ b/.scripts/build_steps.sh
@@ -33,9 +33,9 @@ CONDARC
 
 
 mamba install --update-specs --yes --quiet --channel conda-forge \
-    conda-build pip boa conda-forge-ci-setup=3 "py-lief<0.12"
+    conda-build pip boa conda-forge-ci-setup=3
 mamba update --update-specs --yes --quiet --channel conda-forge \
-    conda-build pip boa conda-forge-ci-setup=3 "py-lief<0.12"
+    conda-build pip boa conda-forge-ci-setup=3
 
 # set up the condarc
 setup_conda_rc "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"

--- a/.scripts/run_osx_build.sh
+++ b/.scripts/run_osx_build.sh
@@ -24,9 +24,9 @@ source ${MINIFORGE_HOME}/etc/profile.d/conda.sh
 conda activate base
 
 mamba install --update-specs --quiet --yes --channel conda-forge \
-    conda-build pip boa conda-forge-ci-setup=3 "py-lief<0.12"
+    conda-build pip boa conda-forge-ci-setup=3
 mamba update --update-specs --yes --quiet --channel conda-forge \
-    conda-build pip boa conda-forge-ci-setup=3 "py-lief<0.12"
+    conda-build pip boa conda-forge-ci-setup=3
 
 
 

--- a/README.md
+++ b/README.md
@@ -1,11 +1,11 @@
-About xeus-sqlite
-=================
+About xeus-sqlite-feedstock
+===========================
+
+Feedstock license: [BSD-3-Clause](https://github.com/conda-forge/xeus-sqlite-feedstock/blob/main/LICENSE.txt)
 
 Home: https://github.com/jupyter-xeus/xeus-sqlite
 
 Package license: BSD-3-Clause
-
-Feedstock license: [BSD-3-Clause](https://github.com/conda-forge/xeus-sqlite-feedstock/blob/main/LICENSE.txt)
 
 Summary: Jupyter kernel for SQLite based on xeus
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -10,7 +10,7 @@ source:
   sha256: 34a42723fec733d42eba8923ee981fc8cf21b88899d851d406bf21aca76975c2
 
 build:
-  number: 0
+  number: 1
   skip: true  # [win]
 
 requirements:
@@ -24,7 +24,8 @@ requirements:
     - zeromq
     - nlohmann_json
     - xtl
-    - cpp-tabulate >=1.3,<2
+    # cpp-tabulate 1.5 is backward incompatible
+    - cpp-tabulate >=1.3,<1.5
     - sqlitecpp
     - sqlite
     - xvega >=0.0.10

